internal/provisioner/equality/equality.go

```diff
@@ -194,38 +194,38 @@ func NodePortServiceChanged(current, expected *corev1.Service) (*corev1.Service,
 	if len(current.Spec.Ports) != len(expected.Spec.Ports) {
 		updated.Spec.Ports = expected.Spec.Ports
 		changed = true
-	}
+	} else {
+		for i, p := range current.Spec.Ports {
+			if !apiequality.Semantic.DeepEqual(p, expected.Spec.Ports[i]) {
+				updated.Spec.Ports = expected.Spec.Ports
+				changed = true
+			}
+		}
```

Comment:
The `else` block should end here -- we should only do the port-by-port comparison if current and expected have the same number of ports, but then all the subsequent comparisons of spec fields should be done regardless.

Reply:
thx, my fault